tests/rptest/wasm/topics_result_set.py

```diff
@@ -32,6 +32,18 @@ def __eq__(self, o):
         return self.topic == o.topic and self.partition == o.partition \
             and self.key == o.key and self.value == o.value
 
+    def __hash__(self):
```

Comment:
I'd recommend using a tuple here (e.g. `return hash((self.topic, self.partition, self.key, etc...))`) (same works for `__eq__`). You could also use a NamedTuple as a drop-in replacement, and store `offset` outside that since it isn't involved in equality. But, it does feel odd that equality doesn't include offset. Should it?

Reply:
Nice catch, i thought i checked that in but it must have gotten lost in a rebase or something.

Reply:
Also about the offset, omitting it was a hack for tests where duplicates could be expected. I think i can provide a cleaner solution for that and include offset in the hash/equality methods